tests/test_spacecurve_casting.py

```diff
@@ -0,0 +1,29 @@
+import jax.numpy as jnp
+import numpy as np
+from qurveros.spacecurve import SpaceCurve
+
+def test_set_params_casts_int_to_float():
+    class DummyCurve(SpaceCurve):
```

Comment:
You should be able to define this `DummyCurve` class just once, outside of these functions.